golink.go

```diff
@@ -541,24 +544,42 @@ func expandLink(long string, env expandEnv) (*url.URL, error) {
 
 func devMode() bool { return *dev != "" }
 
+const peerCapName = "tailscale.com/golink"
+
+type capabilities struct {
+	Admin bool `json:"admin"`
+}
+
+type user struct {
+	login   string
+	isAdmin bool
+}
+
 // currentUser returns the Tailscale user associated with the request.
 // In most cases, this will be the user that owns the device that made the request.
 // For tagged devices, the value "tagged-devices" is returned.
 // If the user can't be determined (such as requests coming through a subnet router),
 // an error is returned unless the -allow-unknown-users flag is set.
-var currentUser = func(r *http.Request) (string, error) {
+var currentUser = func(r *http.Request) (u user, err error) {
 	if devMode() {
-		return "foo@example.com", nil
+		return user{login: "foo@example.com"}, nil
 	}
 	whois, err := localClient.WhoIs(r.Context(), r.RemoteAddr)
 	if err != nil {
 		if *allowUnknownUsers {
 			// Don't report the error if we are allowing unknown users.
-			return "", nil
+			return u, nil
+		}
+		return u, err
+	}
+	u.login = whois.UserProfile.LoginName
```

Comment:
Do we need to handle tagged nodes here? I see we didn't before, and we could in principle grant them access, but I'm not sure if we should. If we do, we will probably want to use `whois.Node.Name` instead, since the user profile is a stub in that case.

Reply:
we rely on that stub user profile (the `tagged-devices` login name) down in `userExists`.  We do allow tagged nodes to create links, but any human user can edit the link and take over ownership.

I suppose this might be another scenario where we could use caps to control this behavior.  You could grant an edit capability to `autogroup:members` and/or `autogroup:tagged` to control whether you want tagged nodes to be able to create links.  We'd need some sensible default behavior if no caps are defined.  I'm not sure that we have a great way to know that today (that no caps are defined with a given name, not just that the *current* peer doesn't have that capability).

Reply:
SGTM. I don't think we have to answer that right now, I just wanted to make sure we weren't going to wind up with weird results. 🙂